### ISSUE TITLE
When using Docker, runOptions `--cpus` should not be set to `1.0` by default

### ISSUE THREAD

**Issue by HarryHung** [nextflow-io/nextflow#3808]
### Expected behavior and actual behavior

When using Docker as the container engine, if `--cpus` is not specify in the `runOptions` and there is no `cpus` directive in the process, it should be default to `--cpus=0` just like the default [docker run](https://docs.docker.com/engine/reference/run/#cpu-period-constraint) instead of `--cpus=1`.

### Steps to reproduce the problem

`main.nf`
```
process ASSEMBLY_SHOVILL {
    input:
    val(n)

    shell:
    '''
    shovill --help
    '''
}

workflow {
    ASSEMBLY_SHOVILL(Channel.from(1))
}
```

`nextflow.config`
```
docker {
    enabled = true
    runOptions = '-u $(id -u):$(id -g)'
}

process {
    container = 'staphb/shovill:1.1.0-2022Dec'
}
```

### Program output 

Excerpt of `.command.run` of the process
```
nxf_launch() {
    docker run -i --cpus 1.0 -v /home/ubuntu/docker-test/work/9d/0139a9416eb085b1e8be02d42a7765:/home/ubuntu/docker-test/work/9d/0139a9416eb085b1e8be02d42a7765 -w "$PWD" -u $(id -u):$(id -g) --name $NXF_BOXID staphb/shovill:1.1.0-2022Dec /bin/bash -ue /home/ubuntu/docker-test/work/9d/0139a9416eb085b1e8be02d42a7765/.command.sh
}
```

### Environment 

* Nextflow version: 22.10.7
* Java version: OpenJDK 17.0.6
* Operating system: Ubuntu 22.04.2, macOS 13.2.1
* Bash version: GNU bash version 5.1.16(1)-release, zsh 5.8.1 (x86_64-apple-darwin22.0)

### Additional context
As mentioned, I would propose the docker run should be default to `--cpus 0.0` or just without this run option if `--cpus` is not specify and the process has no `cpus` directive. 

If this is the intended design, it still have two issues that need to be addressed:
1. When using Singularity as the container engine, the same default cpus limitation is not imposed (it led to a whole day troubleshooting to understand the 4x performance difference between Docker and Singularity using the same VM as local executor 😢). I would suggest their behaviours should be consistent to ensure the portability of Nextflow pipelines. 
2. This implicit default run option is not mentioned in the Nextflow documentation, may be it should as it could have a huge impact on performance.  

Thank you!

**Comment by bentsherman** [comment]
[This commit](https://github.com/nextflow-io/nextflow/commit/b38c38806e11dfdbf32f3fa41287a05f9eb3d430) changed the docker builder to use `--cpu-shares` instead of `--cpus`, which effectively removes the hard limit. It is available since 22.11.0-edge and should be included in 23.04 coming soon. Can you try a more recent edge release and see if that change improves your performance?

**Comment by HarryHung** [comment]
Yep, that fixed it. 😄 
I will just put in a temporary patch for now for my pipeline and wait for the official release of 23.04. 
Thanks!
